src/pages/PastEvents.jsx

```diff
@@ -0,0 +1,177 @@
+import {
+    Box,
+    HStack,
+    Input,
+    InputGroup,
+    InputLeftElement,
+    Heading,
+    Flex,
+    Stack
+  } from '@chakra-ui/react';
+  import { useEffect, useState, useContext } from 'react';
+  import { SearchIcon, HamburgerIcon } from '@chakra-ui/icons';
+  
+  import ImpactSummary from '../components/Events/ImpactSummary';
+  import Backend from '../utils/utils';
+  import Fuse from 'fuse.js';
+  import NavbarContext from '../utils/NavbarContext';
+  import EventsTable from '../components/EventsTable';
+  
+  const PastEvents = () => {
+    const [events, setEvents] = useState([]);
+    const [displayEvents, setDisplayEvents] = useState([]);
+    const [name, setName] = useState('');
+    const [date, setDate] = useState('');
+    const [location, setLocation] = useState('');
+    const [fuse, setFuse] = useState();
+  
+    const { onNavbarDrawerOpen } = useContext(NavbarContext);
+
+    useEffect(() => {
+        console.log('!')
+        console.log(displayEvents);
+    }, [displayEvents])
+  
+    // useEffect(() => {
+    //   // If input is empty, display all volunteers, else conduct the search
+    //   if (input.trim() === '') {
+    //     setDisplayedVolunteers(volunteers);
+    //   } else {
+    //     const options = {
+    //       keys: ['first_name', 'last_name'],
+    //     };
+    //     const fuse = new Fuse(volunteers, options);
+    //     const searchResult = fuse.search(input);
+    //     console.log('search result', searchResult);
+    //     const reduceResult = searchResult.map(result => result.item);
+    //     setDisplayedVolunteers(reduceResult);
+    //   }
+    // }, [input, volunteers]);
+
+    const getEvents = async () => {
+      try {
+        const eventsData = await Backend.get('/events');
+        setEvents(eventsData.data);
+        const options = { keys: ['name', 'date', 'location'], includeScore: true };
+        setFuse(new Fuse(eventsData.data, options));
+      } catch (err) {
+        console.log(`Error getting events: `, err.message);
+      }
+    };
+
+  
+    useEffect(() => {
+      getEvents();
+  
+      // getEventId(eventId);
+    }, []);
+  
+    useEffect(() => {
+      if (!fuse) {
+        return;
+      }
+      console.log(name);
+      let ands = [];
+      if (name) ands.push({ name: name });
+      if (location) ands.push({ location: location });
+      if (date) ands.push({ date: date });
+  
+      let result;
+      if (ands.length > 0) {
+        const fuseResult = fuse.search({ $and: ands });
+        console.log(fuseResult);
+        // If we want to filter by score:
+        // result = fuseResult.filter(item => item.score <= 0.5).map(item => item.item);
+        result = fuseResult.map(item => item.item);
+      } else result = events;
+      console.log(result);
+      setDisplayEvents(result);
+    }, [name, location, date, fuse]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'events'. Either include it or remove the dependency array.